textile/features.textile

```diff
@@ -2405,6 +2411,44 @@ class ReferenceExtras: // REX*
 class ClientInformation: // CR*
   +agents: Dict<String, String?> // CR2
   +agentIdentifier(additionalAgents: Dict<String, String?>?) => String // CR3; interface only offered by some libraries
+
+class BatchResult<T>
+  successCount: number // BAR2a
+  failureCount: number // BAR2b
+  results: [T] // BAR2c
+
+class BatchPublishSpec:
+  channels: [String] // BSP2a
+  messages: [Message] //BSP2b
+
+class BatchPublishSuccessResult:
+  channel: string // BPR2a
+  messageId: string // BPR2b
+
+class BatchPublishFailureResult:
+  channel: string // BPF2a
+  error: ErrorInfo // BPF2c
+
+class BatchPresenceSuccessResult:
+  channel: string // BGR2a
+  presence?: [PresenceMessage] // BGR2b
+
+class BatchPresenceFailureResult
+  channel: string // BGF2a
+  error?: ErrorInfo // BGF2b
```

Comment:
@owenpearson @SimonWoolf is `error` intentionally optional here? Ditto for whether the "if unsuccessful" in BGF2b is redundant.